onecodex/scripts/filter_reads.py

```diff
@@ -0,0 +1,174 @@
+import click
+import csv
+import gzip
+import os
+
+from onecodex.exceptions import OneCodexException, ValidationError
+from onecodex.lib.inline_validator import FASTXTranslator
+from onecodex.utils import download_file_helper, get_download_dest
+
+
+def with_progress_bar(length, ix, *args, **kwargs):
+    label = kwargs.pop('label', None)
+    bar_kwargs = {
+        'length': length
+    }
+    if label:
+        bar_kwargs['label'] = label
+    with click.progressbar(**bar_kwargs) as bar:
+        return ix(cb=bar.update, *args, **kwargs)
+
+
+def filter_rows_by_taxid(results, tax_ids, cb=None):
+    filtered_rows = []
+    for i, row in enumerate(results):
+        if row['Tax ID'] in tax_ids:
+            filtered_rows.append(i)
+        if i % 1000 == 0 and cb:
+            cb(1000)
+    return filtered_rows
+
+
+def get_record_count(fp):
```

Comment:
Does calling `len` on the file object work? If so, would be simpler/cleaner.

Reply:
No, we'd have to convert the file into a list first. But that involves reading the whole file into memory, which I think is sub-optimal since some of these FASTQ's can be GB's. This is just a memory-efficient line counter.

Reply:
That's fine. I think we have a `len` method on our other file which actually reads it through for use with `requests-toolbelt`. Note we should double check that `seek(0)` works properly for our iterators (though I'm sure your tests cover this).

Reply:
Looks like the `len` on the `FASTXTranslator` uses the length of the `Buffer` in bytes, whereas here's we're checking line count. In any case we can definitely move/refactor this in the next round.